### DISCUSSION TITLE
Device 'unreachable'

### DISCUSSION THREAD

**Discussion by JBecker32** [SukramJ/aiohomematic#1377] [Q&A]
Since migrating my old standalone Raspberrymatic to home assistant with raspberrymatic with hahomematic on one raspberry Pi4 I see a strange behaviour of devices becoming 'unreachable'. 
Every few days one of my >30 'switchable' devices (lights) is marked as 'unreachable' in the home assistant UI and also in the Raspberrymatic UI.
In home assistant the last recorded action is always something like 'no longer available by service light turn off' (in german: 'nicht mehr verfügbar ausgelöst durch Dienst light: turn off'). This happens when the device is switched by an automation or manually.
Switching via UI is no longer possible, the device stays in this state forever (more than >20 hours observed ...).
Home assistant log shows:
_2024-01-10 19:02:21.306 WARNING (MainThread) [hahomematic.client] SET_VALUE failed with ClientException [XMLRPC Fault from backend: -1 Failure]: KEQ0927813:1, STATE, False_

BUT: if I go into Raspberrymatic UI, I can solve the problem easily by just 'configuring' the device once. I just go into 'configure' (Einstellen) and say OK without any changes and the device immediately becomes available again.
Does anyone see the same happening and/or have an idea how this can be solved?

**Comment by SukramJ** [comment]
You have a lot of communication problems from your CCU/RM to the devices, which are reported to HA as UNREACH event.
As followup these devices are marked unavailable in HA.

**Solution: Fix your communication problems!** About the howto you should better ask in homematic expert forums. Maybe moving the antenna is enough.

So there is nothing wrong in this integration. 
There is a workaround that is mentioned in the docs to [reactivate devices](https://github.com/danielperna84/custom_homematic#homematicip_localforce_device_availability) and related [blueprints](https://github.com/danielperna84/custom_homematic#available-blueprints).

**Reply by JBecker32** [reply]
There will always be communication problems on 868MHz, true. But these should be solved by retransmission. It does not make sense that **one single** communication error makes the corresponding device 'dead' forever! This would be stupid.
And then it has worked for many years before with my standalone Raspberrymatic installation without one single error like that!

**Reply by SukramJ** [reply]
This is the wrong place for such a discussion. The [RM Repo](https://github.com/jens-maus/RaspberryMatic/discussions) might be better.

**Reply by JBecker32** [reply]
As I said, RM was working absolutly without problems standalone. The problems came with the migration to home assistant ...

**Reply by SukramJ** [reply]
Also mal auf deutsch, das es für dich verständlicher ist:
Dein HomeAssistant-Raspberrymatic.Addon  hat auf einem `PI4` Probleme mit der Gerätekommunikation, die Du vorher nicht bei deiner RaspberryMatic Standalone Installation hattest.
Die Integration meldet jetzt, das Geräte nicht verfügbar sind, die früher mit der Standalone RM Installation verfügbar waren.

==> Die Integration bekommt von der CCU/RM/Addon Benachtichtigungen, das Geräte nicht verfügbar sind.
Warum? Deine RM Addon hat Kommunikationsprobleme mit den Geräten.

Diese Integration kann die Kommunikation der RM zu den Geräten weder verbessern noch verschlechtern, sondern wertet die Informationen nur aus.
Informiere dich in den einschlägigen Homematic-Foren, wie man mit Kommunikationsproblemen zu Geräten  umgehen kann. 
Der PI4 ist bekannt als Funktörer und für RM, ob Addon oder Standalone, nicht die erste Wahl.

**Reply by JBecker32** [reply]
Vielen Dank für Deine Antwort (die ich auch in Englisch verstanden hätte).
Es bleibt für mich merkwürdig, warum ich die Probleme erste jetzt habe. Hardware und Positionierung habe ich nicht verändert.

ich werde jetzt wohl die Homematic Schaltaktoren durch Shellys ersetzen, mit denen habe ich keinerlei Probleme.
Gruß, Jörg.

**Reply by SukramJ** [reply]
Das Problem ist nicht Homematic sondern der PI4.
